### ISSUE TITLE
Spike Pit and Glass Pit Blueprint zone does not build over extant shallow pit

### ISSUE THREAD

**Issue by CSHague** [CleverRaven/Cataclysm-DDA#37297]
# Describe the bug

Attempting to build a construction of a glass or spike pit over a currently existing shallow pit via the construction zone manager does not allow the construction to complete. NPCs given the task will end with "<Name> completed the assigned task" without digging a deep pit to continue the pit construction. With player attempting to build construction plots exits the menu with no response. This only applies with shallow pits when making spike or glass pits as far as I've tested. Building a log wall over an existing shallow pit works fine. If any shallow pit is anywhere in the blueprint zone, the action will fail, even if there are other eligible tiles to be dug.

# Steps To Reproduce

1. Dig a shallow pit with a tool.

2. Designate a spike pit or glass pit over the shallow pit.
3a. Attempt to construct plots with zone action menu. Nothing happens.
3b.  Order recruited NPC to do some construction work. Message log declares "<NPC> has completed assigned task." This is not the case.
# Expected behavior
Should dig a deep pit to move on to making a spike or glass pit.
# Versions and configuration
- OS: Windows
    - OS Version: 10.0 1903
- Game Version: 0.D-11538-gf27d795 [64-bit]
- Graphics Version: Tiles
- Mods loaded: [
    Dark Days Ahead [dda],
    Magiclysm [magiclysm],
    Aftershock [aftershock],
    Bens GF recipes [Tolerate_This],
    Garden Pots [growable-pots],
    Medieval and Historic Classes and Shields [Medieval_Stuff],
    More Survival Tools [More_Survival_Tools],
    Makeshift Items Mod [makeshift],
    Modular Turrets [modular_turrets],
    Salvaged Robots [Salvaged_Robots],
    Hydroponics [hydroponics],
    Mutant NPCs [mutant_npcs],
    Beta National Guard Camp [national_guard_camp],
    More Locations [more_locations],
    Fuji's More Buildings [FujiStruct],
    Folding Parts pack [deoxymod],
    Vehicle Additions Pack [blazemod],
    Roadheader and other mining vehicles [Heavy miners],
    Tanks and Other Vehicles [Tanks]
]

# Additional context

I think the problem might have something to do with Deep Pit being missing as a construction task. While testing, a glass pit will dig to completion over regular terrain. Somewhere the process cannot begin with a shallow pit but must go from dirt to shallow pit construction to deep pit construction but the shallow pit never gets checked as part of the terrain for the next in-progress construction to be placed. However, if the process starts with a shallow pit, it doesn't know where to go because shallow pit to deep pit doesn't exist as a construction but as a vague step between. That's just a guess, however.

**Comment by snipercup** [comment]
I have the same problem except the npc will not construct a spike pit on a deep pit.

This is the terrain:
![image](https://user-images.githubusercontent.com/50166150/72884090-d079de80-3d05-11ea-9ac4-90c18796cffb.png)


I can manually construct the spike pit:
![image](https://user-images.githubusercontent.com/50166150/72884054-be983b80-3d05-11ea-8f19-671b438b6675.png)

The zone:
![image](https://user-images.githubusercontent.com/50166150/72884164-f30bf780-3d05-11ea-9a24-ccf5d835ce00.png)


The NPC never starts, just completes the task with no action taken:
![image](https://user-images.githubusercontent.com/50166150/72884206-0b7c1200-3d06-11ea-8e47-21a61d40b676.png)

**Comment by CSHague** [comment]
@snipercup You have the spears for the pit there? I just tested this in my game and I was able to make spike pits from deep pits in a zone.

**Comment by snipercup** [comment]
Yeah the second screenshot shows the pointy sticks are present. They are on the ground in the camp storage zone. I have not tried using spears. I did not have my player character try to construct the spike pit, I want the npc to do it.

**Comment by CSHague** [comment]
Yeah, that's what I meant, I had an NPC do the same thing with an already-dug deep pit and wooden spears directly at her feet. When she didn't have spears, there would be a delay where she would walk to pit and then "complete the assigned task" but with my shallow pits, the task end message shows up one turn after closing the npc chat box and they make no attempt to move to the spot. Is this behavior identical in this instance? I see the log showing one second difference so it would appear that way but I just wanna confirm.

**Comment by snipercup** [comment]
Ah Okay I get it now, you had the spear directly at the feet. I tried that as well
![image](https://user-images.githubusercontent.com/50166150/72949609-15932480-3d89-11ea-9eab-45d8176712d1.png)

He then made the construction:
![image](https://user-images.githubusercontent.com/50166150/72949628-280d5e00-3d89-11ea-93c5-0ddd1e796594.png)

I did notice I could not drop the pointy sticks on the tile because it was full:
![image](https://user-images.githubusercontent.com/50166150/72949649-38253d80-3d89-11ea-9256-f8e121074b1c.png)

So I had to move some soil first so I could drop the pointy sticks on that tile. I'm guessing the NPC tried to move the pointy sticks but also ran into the issue of not being able to drop it? Actually, that's not the case, he can work with the pointy sticks one tile away:
![image](https://user-images.githubusercontent.com/50166150/72949863-d31e1780-3d89-11ea-95ed-6d80e9e2938c.png)

The tile he is standing on is full (1000L) and the sticks are not moved by the npc. I guess you are right, the sticks have to be close to the construction otherwise the NPC thinks there is no construction material even though the sticks are in the faction camp storage. But then again you could issue a construction order without a faction camp. 

I wouldnt consider this a problem if the NPC told me what is going on but we need  #37300 for that.

**Comment by CSHague** [comment]
I think they will fetch anything that's visible for the task. I watched NPCs walk 12 tiles away to get glass from shattered windows for a glass pit

**Comment by davidpwbrown** [comment]
They will use stuff thats within crafting range already, or go and fetch needed components if they are in a loot zone.
